lib/arborist/build-ideal-tree.js

```diff
@@ -355,6 +359,41 @@ module.exports = cls => class IdealTreeBuilder extends cls {
     })
   }
 
+  async [_retrieveSpecName] (spec) {
+    // if it's just @'' then we reload whatever's there, or get latest
+    // if it's an explicit tag, we need to install that specific tag version
+    const isTag = spec.rawSpec && spec.type === 'tag'
+
+    if (spec.name && !isTag)
+      return spec
+
+    const mani = await pacote.manifest(spec)
+    // if it's a tag type, then we need to run it down to an actual version
+    if (isTag)
+      return npa(`${mani.name}@${mani.version}`)
+
+    spec.name = mani.name
+    return spec
+  }
```

Comment:
This was just a small refactor for readability since when adding the `_followSymlinkPath` method (that is now below) things got gnarly in this part of the code